src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -235,23 +235,17 @@ public async Task<IActionResult> Index(BulkActionViewModel model)
                 return Forbid();
             }
 
-            var allSettings = _shellHost.GetAllSettings();
-
             foreach (var tenantName in model.TenantNames ?? Enumerable.Empty<string>())
             {
-                var shellSettings = allSettings
-                    .Where(x => String.Equals(x.Name, tenantName, StringComparison.OrdinalIgnoreCase))
-                    .FirstOrDefault();
-
-                if (shellSettings == null)
+                if (!_shellHost.TryGetSettings(tenantName, out var shellSettings))
                 {
                     break;
                 }
 
                 switch (model.BulkAction.ToString())
                 {
                     case "Disable":
-                        if (String.Equals(shellSettings.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase))
+                        if (shellSettings.IsDefaultShell())
```

Comment:
We will keep this change or not, see my other and more detailed review comment.

Reply:
I think I already did a PR for this couple months ago, let me find it

Reply:
It's here #11435, so please don't re-invent the wheel, furthermore check @jtkech there ;)

Reply:
@hishamco I see that now. Unfortunately, this change is in 3 different PR which is why I hate when PR site in the queue for so long. So merging any of these 3 PRs, will cause a conflict in two.

Unless you mind, I would like to keep the change here too and when the first PR is merged, we'll just fix this simple conflict.